library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileOutputOperator.java

```diff
@@ -645,6 +647,46 @@ protected FSDataOutputStream openStream(Path filepath, boolean append) throws IO
   }
 
   /**
+   * Opens the stream for the given file path in append mode. Catch the exception if the FS doesnt support
+   * append operation and calls the openStreamForNonAppendFS().
+   * @param filepath given file path
+   * @return output stream
+   */
+  protected FSDataOutputStream openStreamInAppendMode(Path filepath)
+  {
+    FSDataOutputStream fsOutput = null;
+    try {
+      fsOutput = fs.append(filepath);
+    } catch (IOException e) {
+      if (e.getMessage().contains("Not supported")) {
```

Comment:
Error message consists of very huge contents. I think contains() is the best option.